sdk/containerregistry/Azure.Containers.ContainerRegistry/src/Blob/ContainerRegistryBlobClient.cs

```diff
@@ -368,6 +373,98 @@ public virtual async Task<Response<UploadBlobResult>> UploadBlobAsync(Stream str
             }
         }
 
+        private async Task<ChunkedUploadResult> UploadInChunks(string location, Stream stream, int maxChunkSize, bool async = true, CancellationToken cancellationToken = default)
+        {
+            long length = stream.Length;
+
+            string digest = default;
+            ResponseWithHeaders<ContainerRegistryBlobUploadChunkHeaders> uploadChunkResult = null;
+
+            if (length < maxChunkSize)
+            {
+                digest = OciBlobDescriptor.ComputeDigest(stream);
+
+                uploadChunkResult = async ?
+                    await _blobRestClient.UploadChunkAsync(location, stream, cancellationToken: cancellationToken).ConfigureAwait(false) :
+                    _blobRestClient.UploadChunk(location, stream, cancellationToken: cancellationToken);
+
+                return new ChunkedUploadResult(digest,
+                    uploadChunkResult.Headers.Location,
+                    length);
+            }
+
+            byte[] buffer = new byte[maxChunkSize];
+            var ranges = GetRanges(length, maxChunkSize);
+            using SHA256 sha256 = SHA256.Create();
+
+            for (int i = 0; i < ranges.Count; i++)
+            {
+                var range = ranges[i];
+
+                location = uploadChunkResult?.Headers.Location ?? location;
+                var chunkLength = async ?
+                    await stream.ReadAsync(buffer, 0, (int)range.Length, cancellationToken).ConfigureAwait(false) :
+                    stream.Read(buffer, 0, (int)range.Length);
+
+                if (i < ranges.Count - 1)
+                {
+                    sha256.TransformBlock(buffer, 0, chunkLength, buffer, 0);
+                }
+                else
+                {
+                    sha256.TransformFinalBlock(buffer, 0, chunkLength);
+                    digest = "sha256:" + OciBlobDescriptor.BytesToString(sha256.Hash);
```

Comment:
nit: interpolated string

Reply:
Thanks!  Do we have a guideline against these and/or are they bad for perf?  (What do we know?)  What is the recommended action?

Reply:
I don't know if we have specific guidance, but I believe interpolated strings are preferred as the runtime is able to optimize as needed to use stringbuilder. I don't think there is any realistic perf difference - it is more of a style comment.

Reply:
No, that's great.  Thank you!